packages/blade/src/components/Box/__tests__/styled-props.test.ts

```diff
@@ -0,0 +1,44 @@
+import { filterStyledProps, getStyledProps } from '~components/Box/styled-props';
+
+describe('getStyledProps', () => {
+  it('should ignore props that are not styled-props', () => {
+    expect(getStyledProps({ marginTop: 'spacing.2', padding: 'spacing.1' })).toMatchInlineSnapshot(`
+      Object {
+        "marginTop": "spacing.2",
+      }
+    `);
+  });
+
+  it('should print all styled-props', () => {
+    // This test is meant like an alert if we end up adding some prop unexpectedly.
+    // Do update the snapshot if adding new prop was intentional
+    expect(filterStyledProps({})).toMatchInlineSnapshot(`
+      Object {
+        "alignSelf": undefined,
```

Comment:
why is undefined required? shouldn't it return empty object?

Reply:
`getStyledProps` returns undefined. This is internal utility used by getStyledProps. It helps with few things - https://github.com/razorpay/blade/pull/1016#discussion_r1135040604